Core/include/Acts/EventData/VectorMultiTrajectory.hpp

```diff
@@ -0,0 +1,326 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/MultiTrajectory.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+class VectorMultiTrajectory final
+    : public MultiTrajectory<VectorMultiTrajectory> {
+  friend MultiTrajectory<VectorMultiTrajectory>;
+
+  struct IndexData {
+    size_t iprevious = kInvalid;
```

Comment:
Did we settle on types here? I.e. unit32 for all of those?

Reply:
Should be ok to have `uint32_t` here as well I suppose.

Reply:
Should be `uint32_t` consistently now.